src/coreclr/vm/amd64/unixstubs.cpp

```diff
@@ -48,8 +22,43 @@ extern "C"
         return ((eax & 0x06) == 0x06) ? 1 : 0;
     }
 
+#ifndef XSTATE_MASK_AVX512
+#define XSTATE_MASK_AVX512 (0xE0) /* 0b1110_0000 */
+#endif // XSTATE_MASK_AVX512
+
     DWORD avx512StateSupport()
     {
+#if defined(TARGET_OSX)
+        // MacOS has specialized behavior where it reports AVX512 support but doesnt
+        // actually enable AVX512 until the first instruction is executed and does so
+        // on a per thread basis. It does this by catching the faulting instruction and
+        // checking for the EVEX encoding. The kmov instructions, despite being part
+        // of the AVX512 instruction set are VEX encoded and dont trigger the enablement
+        //
+        // See https://github.com/apple/darwin-xnu/blob/main/osfmk/i386/fpu.c#L174
+
+        // TODO-AVX512: Enabling this for OSX requires ensuring threads explicitly trigger
```

Comment:
To me, it sounds like that it is not worth the troubles to enable AVX512 for OSX x64.

Reply:
That's certainly possible. I think we do need some comment tracking/explaining the general reason why its not supported however so the only other change I could see is removing the commented out code.

Reply:
Updated to remove the commented out code. I still left the general TODO, but its likely not something we'll actively focus on supporting given the general move of OSX to Arm64.